app/controllers/api/v1/book_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Api
+  module V1
+    class BookController < ApiController
+      BooksReducer = Rack::Reducer.new(
+        Book.all,
+        ->(author:) { where('lower(author) like ?', "%#{author.downcase}%") },
+        ->(gender:) { where('lower(gender) like ?', "%#{gender.downcase}%") },
+        ->(title:) { where('lower(title) like ?', "%#{title.downcase}%") }
+      )
```

Comment:
### Remember to set a correct base branch because this PR includes changes to another #10 

 * you should use this section in RackReducer documentation https://github.com/chrisfrank/rack-reducer#with-rails-scopes
* and define the filter inside the model and in controller only call the apply method